packages/core/root/core-root-server-internal/src/bootstrap.ts

```diff
@@ -38,15 +42,37 @@ export async function bootstrap({ configs, cliArgs, applyConfigOverrides }: Boot
   // eslint-disable-next-line @typescript-eslint/no-var-requires
   const { REPO_ROOT } = require('@kbn/repo-info');
 
-  const env = Env.createDefault(REPO_ROOT, {
+  let env = Env.createDefault(REPO_ROOT, {
     configs,
     cliArgs,
     repoPackages: getPackages(REPO_ROOT),
   });
 
-  const rawConfigService = new RawConfigService(env.configs, applyConfigOverrides);
+  let rawConfigService = new RawConfigService(env.configs, applyConfigOverrides);
   rawConfigService.loadConfig();
 
+  // Hack to load the extra serverless config files if `serverless: {projectType}` is found in it.
+  const rawConfig = await firstValueFrom(rawConfigService.getConfig$());
+  const serverlessProjectType = rawConfig?.serverless;
+  if (serverlessProjectType) {
+    const extendedConfigs = [
+      ...['serverless.yml', `serverless.${serverlessProjectType}.yml`]
```

Comment:
Will rawConfigService already apply the config-schema validation so that `serverlessProjectType` could only ever be valid serverless project types?

Reply:
`rawConfigService` doesn't apply the validation at this point.

For that we need to create a new instance of the `ConfigService` and inject the schema to it in a similar fashion to `@kbn/cli-dev-mode`: https://github.com/elastic/kibana/blob/34b021c4fe36f5931cf82c0c9d44bff22e2a24ae/packages/kbn-cli-dev-mode/src/config/load_config.ts#L25-L29

However, the validation is performed on Kibana startup.

IMO, we should be fine:

1. If an invalid `serverless` project type is provided, it won't find the file `serverless.${serverlessProjectType}.yml`, so it won't load it.
2. Then, as Kibana's core services are initialized, the `ConfigService` validates the input and throws

Starting `yarn start --serverless=elasticsearch` (after removing the CLI option from `serve.js`):

```
 FATAL  Error: [config validation of [serverless]]: types that failed validation:
- [config validation of [serverless].0]: expected value to equal [es]
- [config validation of [serverless].1]: expected value to equal [oblt]
- [config validation of [serverless].2]: expected value to equal [security]
```

Even if a malicious actor injects a _custom_ project config: `serverless.malicious.yml` and it includes an overwrite of the `serverless` setting:

```yml
# serverless.malicious.yml
serverless: es # a valid project
some.malicious: config
```

This would still fail because `kibana.yml` (and ENV, and CLI) takes precedence over the `serverless.*yml` configs. So `serverless: malicious` will still make it to the validation and throw the error above.